Improving and updating information about Qiskit Code Assistant (#4375)
Qiskit/documentation: docs/guides/qiskit-code-assistant.mdx

# Qiskit Code Assistant

> Qiskit Code Assistant aims to make quantum computing more accessible to new Qiskit adopters and to improve the coding experience for current users. It is a generative AI code assistant powered by [watsonx](https://www.ibm.com/products/watsonx-ai). It is trained using millions of text tokens from Qiskit SDK, years of Qiskit code examples and IBM Quantum&reg; features. Qiskit Code Assistant can help your quantum development workflow by offering LLM-generated suggestions based on [IBM Granite models,](https://www.ibm.com/products/watsonx-ai/foundation-models) which incorporate the latest features and functionalities from IBM&reg;.

Reviewer: ```suggestion
Qiskit Code Assistant aims to make quantum computing more accessible to new Qiskit adopters and to improve the coding experience for current users. It is a generative AI code assistant powered by [watsonx](https://www.ibm.com/products/watsonx-ai). It is trained using millions of text tokens from Qiskit SDK, years of Qiskit code examples, and IBM Quantum&reg; features. Qiskit Code Assistant can help your quantum development workflow by offering LLM-generated suggestions based on [IBM Granite models,](https://www.ibm.com/products/watsonx-ai/foundation-models) which incorporate the latest features and functionalities from IBM&reg;.
```

## The Qiskit HumanEval and Qiskit HumanEval Hard benchmarks

To test the `mistral-small-3.2-24b-qiskit` and other models, we collaborated with Qiskit Advocates and experts to create the execution-based benchmarks called Qiskit HumanEval (QHE) and Qiskit HumanEval Hard (QHE Hard), and ran them on the models. These benchmarks are similar to [HumanEval](https://arxiv.org/abs/2107.03374), including multiple challenging code problems to solve, all based on the official Qiskit libraries.

The benchmarks are composed of approximately 150 tests, each one made from a function definition, followed by a docstring that details the task the model is required to solve. Each example also includes a reference canonical solution, as well as unit tests, to evaluate the correctness of the generated solutions. There are three levels of difficulty for tests: basic, intermediate, and difficult. The Qiskit HumanEval Hard benchmark is a variation of the Qiskit HumanEval one, but removing information related to code imports, so the LLM needs to figure out the right method or class imports. This change makes the dataset to be much more challenging for LLMs, according to our tests and initial results.

> The datasets for Qiskit HumanEval and Qiskit HumanEval Hard are available at https://huggingface.co/datasets/Qiskit/qiskit_humaneval and https://huggingface.co/datasets/Qiskit/qiskit_humaneval_hard. You can contribute to the development of these benchmarks in https://github.com/qiskit-community/qiskit-human-eval.

Reviewer: ```suggestion
The datasets for Qiskit HumanEval and Qiskit HumanEval Hard are available at these websites: [Qiskit HumanEval](https://huggingface.co/datasets/Qiskit/qiskit_humaneval) and [Qiskit HumanEval](https://huggingface.co/datasets/Qiskit/qiskit_humaneval_hard). You can contribute to the development of these benchmarks at the [GitHub repository.](https://github.com/qiskit-community/qiskit-human-eval)
```

## More information and citations

> To learn more about Qiskit Code Assistant, the Qiskit HumanEval or Qiskit HumanEval Hard benchmarks, and cite it in your scientific publications, review these recommended citations:

Reviewer: ```suggestion
To learn more about Qiskit Code Assistant, the Qiskit HumanEval, or Qiskit HumanEval Hard benchmarks, and cite them in your scientific publications, review these recommended citations:
```
